test/Concurrency/Runtime/async_stream.swift

```diff
@@ -415,6 +415,53 @@ class NotSendable {}
         expectTrue(expectation.fulfilled)
       }
 
+      tests.test("finish() behavior with multiple values being awaited") {
```

Comment:
note: there are likely a few more test cases to add (ensure delivery order is expected, etc), but i'll hold off on doing that until getting more feedback on the implementation

Reply:
actually, for the purposes of testing this bug fix, adding delivery ordering tests doesn't seem necessary or directly related.